src/mame/sony_news/news_r4k.cpp

```diff
@@ -0,0 +1,1261 @@
+ *   - Zilog Z8523010VSC ESCC serial interface: emulated (see following)
+ *   - Sony CXD8421Q WSC-ESCC1 serial APbus interface controller: skeleton (ESCC connections, probably DMA, APbus interface, etc. handled by this chip)
+ *   - 2x Sony CXD8442Q WSC-FIFO APbus FIFO/interface chips: partially emulated (handles APbus connections and DMA for sound, floppy, etc.)
+ *   - National Semi DP83932B-VF SONIC Ethernet controller: emulated
+ *   - Sony CXD8452AQ WSC-SONIC3 SONIC Ethernet APbus interface controller: mostly(?) emulated
+ *   - Sony CXD8418Q WSC-PARK3: not fully emulated, but some of the general platform functions may come from this chip (most likely a gate array based on what the PARK2 was in older gen NEWS systems)
+ *   - Sony CXD8403Q DMAC3Q DMA controller: Partially emulated
+ *   - 2x HP 1TV3-0302 SPIFI3 SCSI controllers: Partially emulated, only works with the monitor ROM and NEWS-OS 4 right now.
+ *   - ST Micro M58T02-150PC1 Timekeeper RAM: emulated
+ *  DSC-39 XB Framebuffer/video card:
+ *   - Sony CXD8486Q XB: not emulated (most likely APbus interface)
+ *   - 16x NEC D482235G5 Dual Port Graphics Buffers: not emulated
+ *   - Brooktree Bt468KG220 RAMDAC: not emulated
+ *
+ *  Known issues:
+ *  - Monitor ROM command `ss -r` doesn't show most register values. Not sure what mechanism the APmonitor uses to get the register dump.
+ *    (TLB dump is also broken, but unlike the register dump, the TLB dump is broken on the real NWS-5000X too.
+ *     Use the `mp` monitor command instead for a working version that shows the TLB correctly, both on real hardware and in emulation)
+ *  - Reboot/halt+boot fails because the ESCC FIFO isn't reset properly. For now, the emulator must be hard reset.
+ *    It prints out a message (`esccf0: ai->ai_addr points AProm`) and then tries to allocate and use a different FIFO (which fails)
+ *  - The NetBSD installer has some weird issues (selects SCSI ID 0 regardless of the ID, some serial weirdness that NEWS-OS doesn't have, etc)
+ *
+ *  TODO (in no particular order):
+ *  - More complete floppy support (only supports floppy boot at the moment)
+ *  - NetBSD SCSI support (mostly/entirely changes needed in the SPIFI driver)
+ *  - Sound (lots of undocumented hardware)
+ *  - Parallel I/O
+ *  - Framebuffer (and remaining kb/ms support)
+ *  - APbus expansion slots
+ *  - Triage and fix the known issues mentioned above
+ *  - Full save state support
+ */
+
+#include "emu.h"
+
+#include "cpu/mips/r4000.h"
+#include "machine/ram.h"
+#include "machine/timekpr.h"
+#include "dmac3.h"
+#include "news_hid.h"
+#include "machine/spifi3.h"
+#include "machine/upd765.h"
+#include "cxd8442q.h"
+#include "cxd8452aq.h"
+#include "machine/dp83932c.h"
+#include "machine/z80scc.h"
+#include "bus/rs232/rs232.h"
+
+#include "machine/nscsi_bus.h"
+#include "bus/nscsi/cd.h"
+#include "bus/nscsi/hd.h"
+
+#include "imagedev/floppy.h"
+#include "formats/pc_dsk.h"
```

Comment:
Please separate out the headers from src/mame/sony_news and keep them grouped together, before the headers from src/devices.  Then sort the headers from src/devices to make it easier to see what’s going on.  Put the headers from src/formats after that.